Mathlib/FieldTheory/IntermediateField.lean

```diff
@@ -707,27 +744,34 @@ theorem toSubalgebra_eq_iff : F.toSubalgebra = E.toSubalgebra ↔ F = E := by
   rfl
 #align intermediate_field.to_subalgebra_eq_iff IntermediateField.toSubalgebra_eq_iff
 
-nonrec theorem eq_of_le_of_finrank_le [FiniteDimensional K L] (h_le : F ≤ E)
+theorem eq_of_le_of_finrank_le [hfin : FiniteDimensional K E] (h_le : F ≤ E)
     (h_finrank : finrank K E ≤ finrank K F) : F = E :=
-  toSubalgebra_injective <|
-    Subalgebra.toSubmodule.injective <| eq_of_le_of_finrank_le h_le h_finrank
+  haveI : Module.Finite K E.toSubalgebra := hfin
+  toSubalgebra_injective <| Subalgebra.eq_of_le_of_finrank_le h_le h_finrank
 #align intermediate_field.eq_of_le_of_finrank_le IntermediateField.eq_of_le_of_finrank_le
 
-theorem eq_of_le_of_finrank_eq [FiniteDimensional K L] (h_le : F ≤ E)
+theorem eq_of_le_of_finrank_eq [FiniteDimensional K E] (h_le : F ≤ E)
     (h_finrank : finrank K F = finrank K E) : F = E :=
   eq_of_le_of_finrank_le h_le h_finrank.ge
 #align intermediate_field.eq_of_le_of_finrank_eq IntermediateField.eq_of_le_of_finrank_eq
 
-theorem eq_of_le_of_finrank_le' [FiniteDimensional K L] (h_le : F ≤ E)
+theorem eq_of_le_of_finrank_le' [FiniteDimensional F L] (h_le : F ≤ E)
     (h_finrank : finrank F L ≤ finrank E L) : F = E := by
+  -- can't use `wlog` because the universe level mismatch
+  -- and we have to use `Type _` because it complains about universe level mismatch for `Type*`
+  suffices H {K L : Type _} [Field K] [Field L] [Algebra K L] {F E : IntermediateField K L}
```

Comment:
I think this `suffice` can be a separate lemma (as it was yesterday I think), the name can be the same with `''` or something like that. Also, can you add a docstring mentioning the difference between the various `eq_of_le_of_finrank_le`?

Reply:
I tried to begin with
```
  Subtype.ext_iff.mp <| extendScalars_injective (a₁ := ⟨F, le_rfl⟩) (a₂ := ⟨E, h_le⟩) <|
    eq_of_le_of_finrank_le h_le <| by
      dsimp only
      let E' := extendScalars h_le
      have h1 := finrank_mul_finrank F E' L
```
but Lean complains it can't find `Algebra F E'` so some instance reminders would be necessary. Maybe you could keep the old instance as an unnamed private instance, and then the proof of this new version would be just three lines.

Reply:
I renamed the original `eq_of_le_of_finrank_le'` to `private theorem eq_of_le_of_finrank_le''` and only leave the new `eq_of_le_of_finrank_le'` as a public function.